### DISCUSSION TITLE
nodejs http module failing to parse chunk data and closes the connection

### DISCUSSION THREAD

**Discussion by raghunathkasula** [nodejs/help#4808] [Q&A]
Node Js version : 4.1.1 
Platform: Linux rhel 8

We make use http module of nodejs and create a http server and test upload and download. 
Our client is making use of Transfer-Encoding: chunked for upload, we see below error from http-module of nodejs. Can you please help us what could cause the below parse error. Let us know if you need any other information from us.

**HTTP 3942139: SERVER new http connection
HTTP 3942139: SERVER socketOnParserExecute 229
HTTP 3942139: SERVER socketOnParserExecute 2808
HTTP 3942139: SERVER socketOnParserExecute 4212
HTTP 3942139: SERVER socketOnParserExecute 4212
HTTP 3942139: SERVER socketOnParserExecute 2808
HTTP 3942139: SERVER socketOnParserExecute 1404
HTTP 3942139: SERVER socketOnParserExecute 940
HTTP 3942139: SERVER socketOnParserExecute 2808
HTTP 3942139: SERVER socketOnParserExecute NaN
HTTP 3942139: parse error
HTTP 3942139: server socket close**

**Comment by marco-ippolito** [comment]
can you provide a repro?

**Comment by raghunathkasula** [comment]
Hi,
We are unable to reproduce the issue with our java http client. This issue happens when the http request is made from a real device and http client resides in the device.


we tested with  the later version of node js binary(10.19.0), but issue still happens with below error
NET 24389: onconnection
NET 24389: _read
NET 24389: Socket._read readStart
HTTP 24389: SERVER new http connection
HTTP 24389: SERVER socketOnParserExecute 231
HTTP 24389: SERVER socketOnParserExecute 12582
HTTP 24389: SERVER socketOnParserExecute 1398
HTTP 24389: SERVER socketOnParserExecute 1398
HTTP 24389: SERVER socketOnParserExecute 1398
HTTP 24389: SERVER socketOnParserExecute 1398
HTTP 24389: SERVER socketOnParserExecute 1398
HTTP 24389: SERVER socketOnParserExecute NaN
HTTP 24389: parse error { [Error: Parse Error]
  bytesParsed: 909,
  code: 'HPE_INVALID_CHUNK_SIZE',
  rawPacket:
<Buffer 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ... > }
NET 24389: destroy
NET 24389: close
NET 24389: close handle
NET 24389: has server
NET 24389: SERVER _emitCloseIfDrained
NET 24389: SERVER handle? true   connections? 0
NET 24389: emit close
HTTP 24389: server socket


we share the pcap if is required?
can you please help us to understand the issue..

**Reply by marco-ippolito** [reply]
yes please without a tcp dump it's very impossible

**Comment by JimmyDaddy** [comment]
Why is there a `NaN` chunkSize before the error throws?

```shell
HTTP 24389: SERVER socketOnParserExecute 1398
HTTP 24389: SERVER socketOnParserExecute NaN
HTTP 24389: parse error { [Error: Parse Error]
bytesParsed: 909,
code: 'HPE_INVALID_CHUNK_SIZE',
```

Can you provide a minimum reproducible repo or a gist?
